src/alfasim_sdk/_internal/alfacase/alfacase_to_case.py

```diff
@@ -1742,6 +1762,7 @@ def load_equipment_description(
         "reservoir_inflows": load_reservoir_inflow_equipment_description,
         "heat_sources": load_heat_source_equipment_description,
         "compressors": load_compressor_equipment_description,
+        "leaks": load_leak_equipment_description,
```

Comment:
Did you try to use auto loaders?

Reply:
yup, using load_instance on every ocasion you've pointed in this PR does break some tests. Here's one of the errors outputs (the others are similar):

```console
________________________________ test_convert_alfacase_to_description[CaseDescription] ________________________________

alfacase_to_case_helper = <tests.alfacase.test_alfacase_to_case.alfacase_to_case_helper.<locals>.AlfacaseHelper object at 0x000001C7CC11B7F0>
class_ = <class 'alfasim_sdk._internal.alfacase.case_description.CaseDescription'>
tmp_path = WindowsPath('C:/Users/ramon/AppData/Local/Temp/pytest-of-ramon/pytest-1541/test_convert_alfacase_to_descr84')        

    @pytest.mark.parametrize("class_", ALL_CLASSES_THAT_NEEDS_SCHEMA)
    def test_convert_alfacase_to_description(alfacase_to_case_helper, class_, tmp_path):
        """
        Test to convert Alfacase from all classes of alfasim_core.simulation_models.case_description that needs a Alfacase schema
        """
        alfacase_test_config = ALFACASE_TEST_CONFIG_MAP[class_.__name__]
        file_path = tmp_path / "test_case.alfacase"
        description_obtained = alfacase_to_case_helper.generate_description(
>           alfacase_test_config
        )

..\..\alfasim-sdk\tests\alfacase\test_alfacase_to_case.py:539:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
..\..\alfasim-sdk\tests\alfacase\test_alfacase_to_case.py:118: in generate_description
    return loader(description_document)
..\..\alfasim-sdk\src\alfasim_sdk\_internal\alfacase\alfacase_to_case.py:2010: in load_case_description
    case_values = to_case_values(document, alfacase_to_case_description)
..\..\alfasim-sdk\src\alfasim_sdk\_internal\alfacase\alfacase_to_case.py:675: in to_case_values
    attr_name, function_handle, document
..\..\alfasim-sdk\src\alfasim_sdk\_internal\alfacase\alfacase_to_case.py:690: in execute_loader
    return loader_function(alfacase_content[attr_name])
..\..\alfasim-sdk\src\alfasim_sdk\_internal\alfacase\alfacase_to_case.py:1853: in load_pipe_description
    generate_pipes_description(alfacase_document) for alfacase_document in document
..\..\alfasim-sdk\src\alfasim_sdk\_internal\alfacase\alfacase_to_case.py:1853: in <listcomp>
    generate_pipes_description(alfacase_document) for alfacase_document in document
..\..\alfasim-sdk\src\alfasim_sdk\_internal\alfacase\alfacase_to_case.py:1848: in generate_pipes_description
    case_values = to_case_values(document, alfacase_to_case_description)
..\..\alfasim-sdk\src\alfasim_sdk\_internal\alfacase\alfacase_to_case.py:675: in to_case_values
    attr_name, function_handle, document
..\..\alfasim-sdk\src\alfasim_sdk\_internal\alfacase\alfacase_to_case.py:690: in execute_loader
    return loader_function(alfacase_content[attr_name])
..\..\alfasim-sdk\src\alfasim_sdk\_internal\alfacase\alfacase_to_case.py:1770: in load_equipment_description
    case_description.EquipmentDescription,
..\..\alfasim-sdk\src\alfasim_sdk\_internal\alfacase\alfacase_to_case.py:1649: in _generate_description
    case_values = to_case_values(document, alfacase_to_case_description)
..\..\alfasim-sdk\src\alfasim_sdk\_internal\alfacase\alfacase_to_case.py:675: in to_case_values
    attr_name, function_handle, document
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

attr_name = 'leaks', loader_function = <function load_instance at 0x000001C7CB9B1F28>
alfacase_content = DescriptionDocument(content=YAML({'mass_sources': {'MASS SOURCE': {'fluid': 'fluid_1', 'tracer_mass_fraction': {'value...h('C:/Users/ramon/AppData/Local/Temp/pytest-of-ramon/pytest-1541/test_convert_alfacase_to_descr84/test_case.alfacase'))

    def execute_loader(
        attr_name: str, loader_function: Callable, alfacase_content: YAML
    ) -> Any:
        """
        Execute the loader function passing the YAML content, and return the value converted
        """
        is_a_description_loader = len(inspect.getfullargspec(loader_function).args) == 1

        if is_a_description_loader:
            return loader_function(alfacase_content[attr_name])
        else:
>           return loader_function(key=attr_name, alfacase_content=alfacase_content)
E           TypeError: load_instance() got an unexpected keyword argument 'key'

..\..\alfasim-sdk\src\alfasim_sdk\_internal\alfacase\alfacase_to_case.py:692: TypeError
```

Reply:
yup, tried, it worked in some places, see the most recent commits
Wouldnt be better to create a tech debit and update everything that is possible?